src/analysis/convex/cone.lean

```diff
@@ -635,12 +635,27 @@ lemma inner_dual_cone_eq_Inter_inner_dual_cone_singleton :
   (s.inner_dual_cone : set H) = ⋂ i : s, (({i} : set H).inner_dual_cone : set H) :=
 begin
   simp_rw [set.Inter_coe_set, subtype.coe_mk],
-  refine set.ext (λ x, iff.intro (λ hx, _) _),
-  { refine set.mem_Inter.2 (λ i, set.mem_Inter.2 (λ hi _, _)),
-    rintro ⟨ ⟩,
-    exact hx i hi },
-  { simp only [set.mem_Inter, convex_cone.mem_coe, mem_inner_dual_cone,
-      set.mem_singleton_iff, forall_eq, imp_self] }
+  ext,
+  refine ⟨λ hx, set.mem_Inter₂_of_mem _, by simp⟩,
+  rintros i hi _ ⟨ ⟩,
+  exact hx i hi,
+end
+
+lemma closed_inner_dual_cone : is_closed (s.inner_dual_cone : set H) :=
+begin
+  -- reduce the problem to showing that dual cone of a singleton `{x}` is closed
+  rw inner_dual_cone_eq_Inter_inner_dual_cone_singleton,
+  apply is_closed_Inter,
+  simp_rw [set_coe.forall, subtype.coe_mk],
+  intros x hx,
```

Comment:
```suggestion
  intro x,
```
Since you are using `simp` in the `have`, you can delegate all the coercion nonsense to `simp`.